### PR TITLE
Add dependencies to docker-compose

### PR DESCRIPTION
**Pull request by gosusnp**
## What
The bootloader should run before the other applications.
If the bootloader fails, it is likely a sign that the setup failed and that we may have an unexpected state that is not safe for the other applications.
We should define the bootloader as a dependencies for the applications that depends on it.

Reverts airbytehq/airbyte#19306

## How

Use [conditional depends_on](https://docs.docker.com/compose/compose-file/#depends_on).
* Use docker compose v2 in our tooling
* Update documentation to use docker compose v2
* Update docker compose configs to have services depend on `airbyte-bootloader` terminating successfully

## Release Notes

Starting from version 0.40.27, Airbyte will require docker compose v2.

It means that docker-compose v2 will be required in order to upgrade any Airbyte instance running with docker.
Docker compose v2 should be available by default with most recent docker desktop installations. For linux environments, please refer to the docker documentation https://docs.docker.com/compose/install/linux/.
Note that docker-compose v2 should act as a drop-in replacement for most docker-compose usage.

Following this upgrade, Airbyte will have an explicit check that prevents the application from starting in case of conflicting configuration that would prevent active connections from running properly.

In case of errors upon upgrade, please refer to the troubleshooting documentation https://docs.airbyte.com/troubleshooting/on-deploying

### CODE REVIEW

**Review by colesnodgrass — APPROVED** [review]

**Comment by davinchia** [comment]
@gosusnp I think your changeset makes sense. Can we wait this next week to merge this in? Curious to see what @airbytehq/production-engineering thinks about baking your changes into the runner images instead of having them as part of the workflow

File: docs/deploying-airbyte/on-gcp-compute-engine.md [hunk 1]
```diff
@@ -62,10 +62,8 @@ sudo usermod -a -G docker $USER
 7. Install `docker-compose` on your VM instance by following the below commands in your VM terminal:
 
 ```bash
-sudo apt-get -y install wget
-sudo wget https://github.com/docker/compose/releases/download/1.26.2/docker-compose-$(uname -s)-$(uname -m) -O /usr/local/bin/docker-compose
-sudo chmod +x /usr/local/bin/docker-compose
-docker-compose --version
+sudo apt-get -y install docker-compose-plugin
```
Comment (benmoriceau) [inline docs/deploying-airbyte/on-gcp-compute-engine.md:65]: Do we need to run that on our google compute instance for cart if we want to bump their platform version.

**Review by benmoriceau — COMMENTED** [review]
Do we know how many deployment we might break with that?

File: docs/deploying-airbyte/on-gcp-compute-engine.md (hunk 1 above)
Comment (gosusnp) [reply docs/deploying-airbyte/on-gcp-compute-engine.md:65]: Yes we do

**Comment by gosusnp** [comment]
@davinchia, having docker compose v2 baked into the runner images would be ideal. I'd say the current approach is not great, any tests that requires starting Airbyte within docker needs to add the step to upgrade docker.

**Comment by gosusnp** [comment]
@benmoriceau, I think most server installs of docker have an extra package for docker compose, so I'd expect this change to break all of those.
Users running with a desktop should be fine because compose v2 is embedded since April. (see [announcing-compose-v2-general-availability](https://www.docker.com/blog/announcing-compose-v2-general-availability/))
Also, compose v1 has been marked for deprecation, and it looks like its end of life is planned April 2023.

Any concerns other than communication about this change?

**Comment by davinchia** [comment]
ok @gosusnp let's wait till Monday so someone from @airbytehq/production-engineering can look into creating a new AMI.

@git-phu has done good work stablising this and I want to know what he thinks.

**Review by davinchia — APPROVED** [review]

**Review by sophia-wiley — APPROVED** [review]

**Pull request merged**